### PR TITLE
Separate recordings based on app name

### LINKED ISSUE #3: Separate recordings to their own folders (optionally?)
Copy of https://github.com/marissa999/decky-recorder/issues/30

### PR DESCRIPTION
**Pull request by evanjs**
Resolves #3 

Unless I am missing something, recordings were not previously separated into folders based on app name.
This is in spite of the related issue being closed.

## Notes
Just checked my deck and it looks like I've only been running with the first commit for months now.
I only really ever use rolling recordings, so I haven't tested non-rolling recordings.

There is some cleanup that can be done (IMO) if this is accepted.

Duplicate functionality can be extracted into a function if needed
Log statements can removed if unneeded

___
I mostly wanted to open this before I forget and to gather feedback, see if this is something others want.
If so, either I can clean this up or @safijari can do it in a way they prefer

### CODE REVIEW

**Comment by safijari** [comment]
I can look into it for sure but I feel like this is redundant work at this point with the launch of game recordings integrated into Steam.

**Comment by evanjs** [comment]
Good call
Let's keep an eye on it for now

At the moment, while decky-recorder provides immediate rolling clip saving, it seems like Steam's native recording does not.

## Supported functions
### With background recording
* Long buffers (e.g. 120 mins)
* Mark point of interest in background recording
* Edits, clipping, etc. are intended to be done outside of the game, IIUC
### Without background recording
* Use hotkey to start recording
---
TL;DR — unless I'm missing something, decky recorder does provide some value over Steam native recording at the moment

Short of cleaning this up and getting it merged, I might consider updating locally and building to get the latest changes to tide me over for now.

**Comment by safijari** [comment]
Yeah I just finished writing a script last night for adding that feature. Will be creating a companion plug-in that will bring Steam's recording to parity with recorder

**Comment by safijari** [comment]
Hey, I'm happy to merge this if you make it a togglable option and default it to off. Don't wanna break current behavior.

Also it would be nice if you can make a hardlink for the most recent recording in the main Videos folder similar to how Shotty does it. Makes it easy to find the last clip for quickly sharing
